src/view.rs

```diff
@@ -94,94 +94,97 @@ use crate::{
     id::Id,
     style::{BoxShadowProp, Style, StyleClassRef},
     view_data::ViewStyleProps,
+    views::stack_from_iter,
     EventPropagation,
 };
 
 pub use crate::view_data::ViewData;
 
-pub type AnyWidget = Box<dyn Widget>;
+pub type AnyView = Box<dyn View>;
 
-pub trait View: Sized {
-    fn view_data(&self) -> &ViewData;
-    fn view_data_mut(&mut self) -> &mut ViewData;
+pub trait IntoAnyView {
+    fn any(self) -> AnyView;
+}
 
-    fn id(&self) -> Id {
-        self.view_data().id()
+impl<V: View + 'static> IntoAnyView for V {
+    fn any(self) -> AnyView {
+        Box::new(self)
     }
+}
 
-    /// This method builds the widget described by the view. Implementations may not assume that an
-    /// implicit scope is available. It may differ from the scope the view was created in or may not
-    /// be available at all.
-    fn build(self) -> AnyWidget;
+pub trait IntoView {
+    fn data(&self) -> &ViewData;
 
-    /// Converts this view into the `AnyView` type.
-    fn any(self) -> AnyView
-    where
-        Self: 'static,
-    {
-        AnyView {
-            view: Box::new(self),
-        }
-    }
-}
+    fn data_mut(&mut self) -> &mut ViewData;
 
-/// This is a type which can hold any view.
-pub struct AnyView {
-    view: Box<dyn DynView>,
+    fn into_view(self) -> impl View;
 }
 
-impl View for AnyView {
-    fn view_data(&self) -> &ViewData {
-        self.view.view_data()
+impl<V: View> IntoView for V {
+    fn data(&self) -> &ViewData {
+        self.view_data()
     }
 
-    fn view_data_mut(&mut self) -> &mut ViewData {
-        self.view.view_data_mut()
+    fn data_mut(&mut self) -> &mut ViewData {
+        self.view_data_mut()
     }
 
-    fn build(self) -> AnyWidget {
-        self.view.build()
+    fn into_view(self) -> impl View {
+        self
     }
 }
 
-trait DynView {
-    fn view_data(&self) -> &ViewData;
-    fn view_data_mut(&mut self) -> &mut ViewData;
-    fn build(self: Box<Self>) -> Box<dyn Widget>;
+/// ToView is for converting some types to Floem Views
+pub trait ToView {
+    fn to_view(self) -> impl View;
 }
 
-impl<V: View> DynView for V {
-    fn view_data(&self) -> &ViewData {
-        self.view_data()
+impl ToView for () {
+    fn to_view(self) -> impl View {
```

Comment:
This should then return `impl IntoView` as end users operate on `IntoView` and we can't assume that all `IntoView` are also `View`. This would effectively be `IntoIntoView` with your rename. That does suggest that the `View` to `IntoView` rename is a bad idea.